### PR TITLE
Support mocked server in testing

### PR DESCRIPTION
**Pull request by wyTrivail**
so that we can use it in pr build to perform end-to-end test by using a fake credentials

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,113 +1,133 @@
...
 
-* [go module to add your component](https://github.com/aws-observability/aws-otel-collector/blob/main/go.mod)
-* [enable your component](https://github.com/aws-observability/aws-otel-collector/blob/main/pkg/defaultcomponents/defaults.go).
-* [link your testcase](https://github.com/aws-observability/aws-otel-collector/blob/main/e2etest/testcases.json)
+You will need to create a sub folder under [the testcase directory](https://github.com/aws-observability/aws-otel-test-framework/tree/terraform/terraform/testcases), and place your configuration into it. Typically, 
+you will need to place files as following(please use the same filename as below):
 
-## 2. Run and debug your testcase
+1. `otconfig.tpl`: which contain the new component, will be used as the config in all types of the e2etest. 
+2. [optional] `parameters.tfvars`: you can run your test case locally with adding `-var-file=../testcases/yourtestcase/parameters.tfvars` to the terraform command. By using this file, you can override the default configuration in the testing framework. [The parameters you can override](terraform/README.md).
```
Comment (mxiamxia) [inline README.md]: I think we should make `parameters.tfvars` file as optional. Should we require each testcase need to be able to run locally as mandatory requirement?

File: README.md [hunk 2]
```diff
@@ -1,113 +1,133 @@
...
-please check https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
+When you link your testcase, there are five types of testing platforms you can choose,
+* LOCAL, which will run the test in the pr workflow. 
+* EC2, which will run the test in the main branch workflow under an ec2 instance.
+* ECS, which will run the test in the main branch workflow under an ecs cluster.
+* EKS, which will run the test in the main branch workflow under an eks/k8s cluster.
+* SOAKING, which will run every night, and perform high throughput to your test case and monitor the resource usages.
 
-#### 2.1.2 install terraform
+Typically, we required all the types for every test case, if you have a special usage of your component which might not need a certain platform,
```
Comment (mxiamxia) [inline README.md]: a little confusing phrase. can we say "we require the new components to be tested on all platforms for all the test cases."

File: README.md [hunk 3]
```diff
@@ -1,113 +1,133 @@
...
+please put the reason in the pr.
+If you find the current test case option can't fulfill your testing requirement, feel free to open an issue, so we can discuss together.
 
-please check https://learn.hashicorp.com/tutorials/terraform/install-cli
+## 3. Run test cases in AWS platform
+
+The commands in Quick Start is to run the test case in containers locally.
+In the case that you want to debug for a certain platform, you can also use this testing framework to run your test case locally in multiple AWS platforms including EC2, ECS and EKS.
+
```
Comment (mxiamxia) [inline README.md:76]: can we provide one example command here? Eg,  run test case on AWS ECS?

File: terraform/mock/variables.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+variable "region" {
+  default = "us-west-2"
+}
+
+variable "validation_config" {
+  default = "default-mocked-server-validation.yml"
+}
+
+variable "data_emitter_image" {
+  default = "josephwy/integ-test-emitter:alpine"
```
Comment (mxiamxia) [inline terraform/mock/variables.tf:27]: can we move this test emitter code into this repo?

File: terraform/testcases/otlp_mock/otconfig.tpl [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  awsemf:
+    region: '${region}'
+    no_verify_ssl: false
+    endpoint: "${mock_endpoint}"
+
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      exporters: [logging, awsemf]
```
Comment (mxiamxia) [inline terraform/testcases/otlp_mock/otconfig.tpl:28]: can we use the same metric pipeline config from our default collector configuration files?

File: terraform/testcases/xrayreceiver_mock/otconfig.tpl [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    local_mode: true
+    no_verify_ssl: false
+    endpoint: "${mock_endpoint}"
+
+service:
+  extensions:
+  pipelines:
+    traces:
+      receivers: [awsxray]
+      exporters: [logging, awsxray]
```
Comment (mxiamxia) [inline terraform/testcases/xrayreceiver_mock/otconfig.tpl:20]: the same here, copy the same trace pipeline config from the default config.

File: terraform/testcases/xrayreceiver_mock/parameters.tfvars [hunk 7]
```diff
@@ -0,0 +1,4 @@
+validation_config="default-mocked-server-xrayreceiver-validation.yml"
+
+# sample application image to emit the trace data
+data_emitter_image="johnwu20/sample_server_app:v2.0.0"
```
Comment (mxiamxia) [inline terraform/testcases/xrayreceiver_mock/parameters.tfvars:4]: just to make sure we maintain all the sample app in one repo?

File: validator/src/main/java/com/amazon/aoc/validators/MockedServerValidator.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+    RetryHelper.retry(
+        () -> {
+          Response response = client.newCall(request).execute();
+          if (!response.isSuccessful()) {
+            throw new BaseException(ExceptionCode.MOCKED_SERVER_NOT_AVAILABLE);
+          }
+
+          String responseBody = response.body().string();
+          if (!responseBody.equalsIgnoreCase("true")) {
```
Comment (mxiamxia) [inline validator/src/main/java/com/amazon/aoc/validators/MockedServerValidator.java]: maybe change "true" to "success" :)

File: README.md [hunk 9]
```diff
@@ -1,113 +1,133 @@
...
 
+### 3.2 Run in ECS
 
-### 2.2 Run Test locally
-
-#### 2.2.1 ECS
-
-```shell
+````
 cd terraform/ecs && terraform init && terraform apply -var="aoc_image_repo={{the docker image repo name you just pushed}}" -var="aoc_version={{ the docker image tag name}} -var="testcase=../testcases/{{your test case folder name}}" -var-file="../testcases/{{your test case folder name}}/parameters.tfvars"
```
Comment (mxiamxia) [inline README.md]: -var="aoc_version={{ the docker image tag name}} -var

missing `"` after `}}`

File: terraform/mock/variables.tf (hunk 4 above)
Comment (wyTrivail) [reply terraform/mock/variables.tf:27]: the sample app code is in the sdk repo.

File: terraform/testcases/otlp_mock/otconfig.tpl (hunk 5 above)
Comment (wyTrivail) [reply terraform/testcases/otlp_mock/otconfig.tpl:28]: yes we can

File: terraform/testcases/xrayreceiver_mock/otconfig.tpl (hunk 6 above)
Comment (wyTrivail) [reply terraform/testcases/xrayreceiver_mock/otconfig.tpl:20]: okay

File: terraform/testcases/xrayreceiver_mock/parameters.tfvars (hunk 7 above)
Comment (wyTrivail) [reply terraform/testcases/xrayreceiver_mock/parameters.tfvars:4]: yes we will, this is a temp one

File: validator/src/main/java/com/amazon/aoc/validators/MockedServerValidator.java (hunk 8 above)
Comment (wyTrivail) [reply validator/src/main/java/com/amazon/aoc/validators/MockedServerValidator.java]: okay

File: README.md (hunk 9 above)
Comment (wyTrivail) [reply README.md]: thx

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**